File: frontend/src/api/match.ts
const API_BASE = '/api'

export interface Match {
  id: string
  userAId: string
  userBId: string
  status: string
  matchedAt: string
}

export const matchApi = {
  getMyMatch: async (userId: string): Promise<Match | null> => {
    const response = await fetch(`${API_BASE}/match/me?userId=${userId}`)
    if (!response.ok) return null
    return response.json()
  },

  createMatch: async (userId: string): Promise<Match> => {
    const response = await fetch(`${API_BASE}/match/create`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ userId })
    })
    if (!response.ok) throw new Error('Failed to create match')
    return response.json()
  }
}

File: backend/Models/Match.cs
namespace MBTIMatch.Models;

public class Match
{
    public Guid Id { get; set; }
    public Guid UserAId { get; set; }
    public Guid UserBId { get; set; }
    public string Status { get; set; } = "pending";
    public DateTime MatchedAt { get; set; }
}
